shipLHC/AdvTarget.cxx

```diff
@@ -247,24 +247,25 @@ Bool_t AdvTarget::ProcessHits(FairVolume *vol)
 
         TParticle *p = gMC->GetStack()->GetCurrentTrack();
         Int_t pdgCode = p->GetPdgCode();
-        TLorentzVector Pos;
-        gMC->TrackPosition(Pos);
+        TLorentzVector ExitPoint;
+        gMC->TrackPosition(ExitPoint);
         TLorentzVector Mom;
         gMC->TrackMomentum(Mom);
         Int_t detID = 0;
         gMC->CurrentVolID(detID);
         fVolumeID = detID;
-        Double_t xmean = (fPos.X() + Pos.X()) / 2.;
-        Double_t ymean = (fPos.Y() + Pos.Y()) / 2.;
-        Double_t zmean = (fPos.Z() + Pos.Z()) / 2.;
+        Double_t xmean = (EntryPoint.X() + ExitPoint.X()) / 2.;
+        Double_t ymean = (EntryPoint.Y() + ExitPoint.Y()) / 2.;
+        Double_t zmean = (EntryPoint.Z() + ExitPoint.Z()) / 2.;
         AddHit(fTrackID,
                fVolumeID,
                TVector3(xmean, ymean, zmean),
                TVector3(fMom.Px(), fMom.Py(), fMom.Pz()),
                fTime,
                fLength,
                fELoss,
-               pdgCode);
+               pdgCode,
+               TVector3(ExitPoint.X(), ExitPoint.Y(), ExitPoint.Z()));
```

Comment:
weren't we also storing the entry point?
In the end, we'd like to minimize calculations on coordinates, so we discussed storing mean position(as in the past), and add the entry and exit positions.

Reply:
To double precision, we can calculate the exact entry point from mean and exit.
This should be good enough for all practical purposes.

IMHO, the only reason for storing also the entry point would be in case the simple calculation is a performance bottleneck, bit it almost certainly isn't.